polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -166,11 +167,14 @@ bool isCandidateKernel(gpu::GPUFuncOp kernel) {
   // TODO: check uniformity.
 }
 
-/// A function is a candidate iff it has an nd_item argument.
+/// A function is a candidate iff it has an sycl::nd_item or sycl::item
+/// argument.
 bool isCandidateFunction(FunctionOpInterface func) {
-  // TODO: construct nd_item when not passed in.
-  if (func.getNumArguments() == 0 ||
-      !sycl::isPtrOf<sycl::NdItemType>(func.getArgumentTypes().back()))
+  if (func.getNumArguments() == 0)
+    return false;
+  Type lastArgTy = func.getArgumentTypes().back();
+  if (!sycl::isPtrOf<sycl::NdItemType>(lastArgTy) &&
+      !sycl::isPtrOf<sycl::ItemType>(lastArgTy))
```

Comment:
Can we get a `sycl.id` here too?

Reply:
Likely we can, but the memory access analysis doesn't support that yet. How about we extent it further in another PR?

Reply:
added.